### PR TITLE
Fixes for cysignal and python3

### PR DESCRIPTION
**Pull request by lducas**
Avoid issue with cysignals by only setting up compilation flag after its installation.
Plus a bootstrap3.sh variant that force python=python3 for the transitional period.

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+jobs="-j 4 "
```
Comment (malb) [inline bootstrap.sh:3]: I don't like hardcoding -j4 as the default. This can overwhelm small boxes.

File: bootstrap3.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+
+jobs="-j 4 "
+if [ "$1" = "-j" ]; then
+   jobs="-j $2 "
+fi
+
+# Create Virtual Environment
+alias python=/usr/bin/python3
```
Comment (malb) [inline bootstrap3.sh:10]: This is the only change compared to bootstrap, right? I think this invites subtle problems. I'd add a parameter to bootstrap "py3" or so which sets this instead.

File: rebuild.sh [hunk 3]
```diff
@@ -5,8 +5,8 @@ enable_stats=0
 enable_ndebug=0
 maxsievingdim=128
 enable_ggdb=0
-enable_jobs=0
 enable_templated_dim=0
+jobs=4
```
Comment (malb) [inline rebuild.sh:9]: Same as above.

File: bootstrap3.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+pip install -r requirements.txt
+pip install -r suggestions.txt
+python setup.py clean
+python setup.py build_ext $jobs
+python setup.py install
+cd ..
+
+pip install -r requirements.txt
+python setup.py clean
+python setup.py build_ext $jobs --inplace
```
Comment (malb) [inline bootstrap3.sh:71]: Does this actually work, i.e. parallelising `build_ext`?

File: bootstrap.sh (hunk 1 above)
Comment (cr-marcstevens) [reply bootstrap.sh:3]: Note that hardcoded compilation with 4 jobs for dependencies and kernel was already in `bootstrap.sh`, my last merged PR just also applied it to python/Cython compilation and `rebuild.sh`.

Comment (cr-marcstevens) [reply bootstrap.sh:3]: See https://github.com/fplll/g6k/blob/d92c5df01fdb91047b185c5134ebb7dfa86888eb/bootstrap.sh

File: bootstrap3.sh (hunk 4 above)
Comment (cr-marcstevens) [reply bootstrap3.sh:71]: Yes, it's actually a nice speedup.

**Comment by lducas** [comment]
Hi, most of the discussion was about jobs, but the key modification was delaying the setup of compilation flags so that cysignals doesn't choke. Are you cool with that ?

**Comment by cr-marcstevens** [comment]
Does activate twice work properly? Otherwise there is also deactivate to undo the first before the second call.

**Comment by lducas** [comment]
There is no separate script to deactive as far as I see, but activate does contain a deactivate function, which is executed at the beginning. Not sure it does all it should do. I did not experienced any issue, on my setup.

**Comment by malb** [comment]
> Hi, most of the discussion was about jobs, but the key modification was delaying the setup of compilation flags so that cysignals doesn't choke. Are you cool with that ?

Yep, all of that looks good to me. +1 for merging from me.

**Pull request merged**